dodiscover/cd/base.py

```diff
@@ -0,0 +1,54 @@
+from abc import ABCMeta, abstractmethod
+from typing import Set, Tuple
+
+import pandas as pd
+
+from dodiscover.typing import Column
+
+
+class BaseConditionalDiscrepancyTest(metaclass=ABCMeta):
+    """Abstract class for any conditional discrepancy test.
+
+    All CD tests are used in constraint-based causal discovery algorithms. This
+    class interface is expected to be very lightweight to enable anyone to convert
+    a function for CD testing into a class, which has a specific API.
+    """
+
+    def _check_test_input(
+        self,
+        df: pd.DataFrame,
+        x_vars: Set[Column],
+        y_vars: Set[Column],
+        group_col: Column,
+    ):
+        if any(col not in df.columns for col in x_vars):
+            raise ValueError("The x variables are not all in the DataFrame.")
+        if any(col not in df.columns for col in y_vars):
+            raise ValueError("The y variables are not all in the DataFrame.")
+        if group_col not in df.columns:
+            raise ValueError(f"The group column {group_col} is not in the DataFrame.")
+
+    @abstractmethod
+    def test(
```

Comment:
Why not use same comments as [here](https://github.com/py-why/dodiscover/pull/83/files#diff-582bc7bbcf4a59801258f952989ed4f625cbbc62f48db75d3b41f28893b28ad6R100)

Reply:
That directory handles CI testing, which tests: `H_0: P(y | x, z) = P(y | z)`, where typically `x,y,z` have to of the same variable type (i.e. binary, discrete, continuous).

while conditional discrepancy tests: `H_0: P(y | x) = P'(y | x)`, or formulated as `H_0: P(y|x,t) = P(y|x)`, where `t` is discrete and `y,x` can be continuous.

I guess they could... be formulated as the same arguments in the API. Perhaps I can refactor the code if there is a lot of similarity between `ci` and `cd` tests to a common just `conditional_testing` submodule.